### PR TITLE
[SDK generation pipeline] allow generate from typespec and swagger at same time

### PR DESCRIPTION
**Pull request by msyyc**
for https://github.com/Azure/azure-sdk-tools/issues/8712

allow generate SDK from typespec and swagger at same time.

test link: https://github.com/test-repo-billy/azure-rest-api-specs/pull/3426

### CODE REVIEW

File: tools/azure-sdk-tools/packaging_tools/sdk_generator.py [hunk 1]
```diff
@@ -273,23 +273,22 @@ def main(generate_input, generate_output):
     result = {}
     python_tag = data.get("python_tag")
     package_total = set()
-    spec_word = "readmeMd"
+    readme_files = []
```
Comment (raych1) [inline tools/azure-sdk-tools/packaging_tools/sdk_generator.py:276]: Seems this array is used to keep both readme files and typespec folder paths. Can we rename it appropriately?

File: tools/azure-sdk-tools/packaging_tools/sdk_generator.py [hunk 2]
```diff
@@ -273,23 +273,22 @@ def main(generate_input, generate_output):
...
         if "specification" in spec_folder:
             spec_folder = str(Path(spec_folder.split("specification")[0]))
         if "specification" not in input_readme:
             input_readme = str(Path("specification") / input_readme)
-        readme_files = [input_readme]
-    else:
+        readme_files.append(input_readme)
+    if "relatedTypeSpecProjectFolder" in data:
         # ["specification/confidentialledger/ConfientialLedger"]
         if isinstance(data["relatedTypeSpecProjectFolder"], str):
```
Comment (raych1) [inline tools/azure-sdk-tools/packaging_tools/sdk_generator.py:288]: `data["relatedTypeSpecProjectFolder"]` should be list type instead of string type?

Comment (msyyc) [reply tools/azure-sdk-tools/packaging_tools/sdk_generator.py:288]: It is compatible logic added by dongwei before and I am not sure whether now the input data will never be string anymore.

File: tools/azure-sdk-tools/packaging_tools/sdk_generator.py (hunk 1 above)
Comment (msyyc) [reply tools/azure-sdk-tools/packaging_tools/sdk_generator.py]: Make sense and rename it to `readme_and_tsp`

File: tools/azure-sdk-tools/packaging_tools/sdk_generator.py (hunk 2 above)
Comment (raych1) [reply tools/azure-sdk-tools/packaging_tools/sdk_generator.py:288]: It's always a string array as the following schema. BTW, `relatedReadmeMdFile` was deprecated and can be deleted.
```typescript
 "relatedReadmeMdFiles": {
      // Related readme.md files that pending generation.
      "type": "array",
      "items": {
        "type": "string"
      }
    },
    "relatedTypeSpecProjectFolder": {
      // Related typespec project folder that pending generation.
      "type": "array",
      "items": {
        "type": "string"
      }
    },
```

Comment (msyyc) [reply tools/azure-sdk-tools/packaging_tools/sdk_generator.py]: Updated.

**Review by raych1 — APPROVED** [review]

**Pull request merged**